setup.py

```diff
@@ -29,24 +14,24 @@
     package_data={"spidermon": ["VERSION"]},
     zip_safe=False,
     include_package_data=True,
-    install_requires=["jsonschema[format]>=3.2.0", "python-slugify"],
-    tests_require=test_requirements,
+    install_requires=[
+        "boto",
+        "boto3",
+        "itemadapter",
+        "Jinja2",
+        "jsonschema[format]>=3.2.0",
+        "premailer",
+        "python-slugify",
+        "Requests",
+        "scrapinghub",
+        "scrapinghub-entrypoint-scrapy",
+        "scrapy",
+        "sentry_sdk",
+        "slack_sdk",
+        "Twisted",
+    ],
     extras_require={
-        # Specific monitors and tools to support notifications and reports
-        "monitoring": [
-            "scrapy",
-            "Jinja2",
-            "scrapinghub",
-            "slack-sdk",
-            "boto",
-            "boto3",
-            "premailer",
-            "sentry-sdk",
-        ],
-        # Tools to run the tests
-        "tests": test_requirements,
-        # Tools to build and publish the documentation
-        "docs": ["sphinx", "sphinx-rtd-theme", "s3cmd"],
+        "monitoring": [],
```

Comment:
Would it make sense to move packags like slack_sdk, sentry_sdk and boto here? Are they optional dependencies of spidermon, or are they all required for spidermon to work?

Reply:
They are not required by Spidermon itself but by different actions:
- boto is required by `actions/reports/s3.py`
- boto3 is required by `actions/email/ses.py`
- sentry_sdk is required by `actions/sentry/__init__.py`
- slack_sdk is required by `actions/slack/__init__.py`

Before this change, most-common user case of spidermon was to install everything with `spidermon[monitoring]`. The idea was to make this the default installing by moving all the packages into `install_requires` and keep `monitoring` empty to be backward compatible. 

Alternatively, we could move this packages into a new section named `actions` but that wouldn't be backwards compatible and may be quite problematic for most people.

Reply:
What about the reqs that were moved from `tests_require` / `spidermon[tests]` to `install_requires`?

Reply:
If I'm not mistake those would be: 

- itemadapter: [[contrib/scrapy/pipelines.py](https://github.com/scrapinghub/spidermon/blob/e6509dd0523acded0d6cf7dd3422a9581a99815a/spidermon/contrib/scrapy/pipelines.py#L4)], [[contrib/scrapy/extensions.py](https://github.com/scrapinghub/spidermon/blob/e6509dd0523acded0d6cf7dd3422a9581a99815a/spidermon/contrib/scrapy/extensions.py#L5)], 
- requests [[contrib/actions/telegram/__init__.py](https://github.com/scrapinghub/spidermon/blob/e6509dd0523acded0d6cf7dd3422a9581a99815a/spidermon/contrib/actions/telegram/__init__.py#L3)],[[contrib/actions/discord/__init__.py](https://github.com/scrapinghub/spidermon/blob/e6509dd0523acded0d6cf7dd3422a9581a99815a/spidermon/contrib/actions/discord/__init__.py#L5)]: 
- scrapinghub-entrypoint-scrapy (sh_scrapy) [[contrib/stats/statscollectors/sc_collections.py](https://github.com/scrapinghub/spidermon/blob/e6509dd0523acded0d6cf7dd3422a9581a99815a/spidermon/contrib/stats/statscollectors/sc_collections.py#L6)]
- twisted [[contrib/scrapy/extensions.py](https://github.com/scrapinghub/spidermon/blob/e6509dd0523acded0d6cf7dd3422a9581a99815a/spidermon/contrib/scrapy/extensions.py#L4)]

itemadapter, requests and sh_scrapy are required by different parts of the code. I guess they were not previously added by mistake, because they are already present on most Scrapy stacks anyway. 

The only package that seems redundant to me is twisted which I think we should drop as is already covered by `scrapy`.

Reply:
I assume the dependencies like boto or slack-sdk are all optional, and most users don't use all of them. 

Not installing them by default minimizes scenarios when some of these packages don't install for some reason for the users - e.g. a package drops support for some Python version which spidermon still supports (and which an user is using), or requirements of some package conflict with requirements which user's project has. 

If these packages are not optional, then in such scenario spidemon installation is broken for users who don't even use these features.

That said, if based on your experience with these particular packages you found it unlikely they break, it looks fine to install them by default.

So, I'm approving the PR, and let you folks decide if the concern is worth addressing, or if addressing it just complicates things.

If it's something which should be addressed, probably the easiest way to go is to move all optional requirements to some section like `pip install spidermon[contrib]` or `pip install spidermon[all]`, keep only the required ones in default install_requires, and also document `pip install spidermon[all]` as the way to install spidermon.

Reply:
I'm worried about adding a new optional requirements like `contrib` all `all` after we already have `monitors` in place that is used by default by most users.